src/Core/src/Hosting/LifecycleEvents/AppHostBuilderExtensions.cs

```diff
@@ -1,36 +1,34 @@
 ﻿using System;
 using Microsoft.Extensions.DependencyInjection;
-using Microsoft.Extensions.Hosting;
-using Microsoft.Maui.Hosting;
 
 namespace Microsoft.Maui.LifecycleEvents
 {
-	public static partial class AppHostBuilderExtensions
+	public class LifecycleEventRegistration
 	{
-		public static IAppHostBuilder ConfigureLifecycleEvents(this IAppHostBuilder builder, Action<ILifecycleBuilder> configureDelegate)
-		{
-			builder.ConfigureServices<LifecycleBuilder>((_, lifecycle) => configureDelegate(lifecycle));
+		private readonly Action<ILifecycleBuilder> _registerAction;
 
-			return builder;
+		public LifecycleEventRegistration(Action<ILifecycleBuilder> registerAction)
+		{
+			_registerAction = registerAction;
 		}
 
-		public static IAppHostBuilder ConfigureLifecycleEvents(this IAppHostBuilder builder, Action<HostBuilderContext, ILifecycleBuilder> configureDelegate)
+		internal void AddRegistration(ILifecycleBuilder effects)
 		{
-			builder.ConfigureServices<LifecycleBuilder>(configureDelegate);
-
-			return builder;
+			_registerAction(effects);
 		}
+	}
 
-		class LifecycleBuilder : LifecycleEventService, ILifecycleBuilder, IMauiServiceBuilder
+	public static partial class MauiAppHostBuilderExtensions
+	{
+		public static MauiAppBuilder ConfigureLifecycleEvents(this MauiAppBuilder builder, Action<ILifecycleBuilder>? configureDelegate)
 		{
-			public void ConfigureServices(HostBuilderContext context, IServiceCollection services)
+			builder.Services.AddSingleton<ILifecycleEventService, LifecycleEventService>();
```

Comment:
One thing that I am noticing with the new way is that this service is getting registered multiple times. Is that going to cause issues?

```csharp
builder.ConfigureLifecycleEvents(app => app.Xxx());
...
builder.ConfigureLifecycleEvents(app => app.Xxx());
```

Each call to `ConfigureLifecycleEvents` will re-register `LifecycleEventService`

Reply:
Yeah most (or all?) of the framework calls I need to change to TRY AddXyz and that will fix this.